### PR TITLE
Clean up the Challenge Protocol spec to be easier to edit by humans

### PR DESCRIPTION
**Pull request by mcy**

### CODE REVIEW

**Review by alistair23 — APPROVED** [review]

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: We should move this to a table.  It doesn't have to look like the table in the original spec, but I think some kind of table representation would be more generally consumable.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: Similarly here, we should go back to some kind of table.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: This doesn't seem to render well in github's viewer.  Is that a viewer-specific issue or there something else going on here.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: Another instance where I think a table is preferred.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: Needs an extra back-tick after Update

**Review by chweimer — CHANGES_REQUESTED** [review]
This looks great.  Thanks for all the work on it.  Just a few minor comments, most of which we had already talked about.

File: RoT/Protocol/Challenge_Protocol.md
Comment (mcy) [reply RoT/Protocol/Challenge_Protocol.md]: This was a formatting mistake on my part. Good catch.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: GET_DIGESTS

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: requested

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: This conversion lost some detail about this 'rq' bit.  It doesn't really distinguish between request/response messages.

From the original text:
Request Type
   </td>
   <td>This field indicates what type of request is contained in the message.  Messages defined in this specification shall have this bit set to 0.  Setting this bit to 1 provides a mechanism, aside from different vendor IDs, to support a device-specific command set.  Devices that don’t have any additional command support will return an error if this bit is 1.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: Messages are not always encrypted.

File: RoT/Protocol/Challenge_Protocol.md
Comment (chweimer) [inline RoT/Protocol/Challenge_Protocol.md]: Same as the comment in the previous section.

**Comment by chweimer** [comment]
This is a lot of good work to reformat things for easier editing and diffs, but I did notice a couple places (the MCTP rq bit) where some information got lost.  I haven't read through the entire modified spec in detail to see if there are other places like this.  We have a couple options:
1. I read through in detail and see if I see anything else that is misrepresented.
2. We complete the PR as-is with the intended goal of reformatting the markdown.  I can then take an follow-up action to read through and edit an necessary to match the current spec.

Thoughts or preferences?

**Comment by mcy** [comment]
I think that depends on how much time you want to spend on it; some of the typos you pointed out are original, I think, and I plan to go through the whole thing and clean it up. (That said, I will fix what you've pointed out; I've definitely made mistakes copy/pasting tables around.)

Selfishly, I would prefer to get this in sooner, because then we can start doing further cleanups in parallel.

**Comment by chweimer** [comment]
From my perspective, either way I'm going to need to go through it.  It would probably be best to just complete this as-is with another round of cleanup as a follow-up.  It would probably also be easier to just directly edit anything that may be a mistake instead of going through PR comments.

**Comment by mcy** [comment]
Sounds good. I'll let you approve when you're comfortable with the state of the PR, and we can go from there.

**Comment by mcy** [comment]
Sounds good. I'll let you approve when you're comfortable with the state of the PR, and we can go from there.

**Comment by chweimer** [comment]
I think if we just clean up the few things that I've already noted, that would be good enough for a first pass.

File: RoT/Protocol/Challenge_Protocol.md
Comment (mcy) [reply RoT/Protocol/Challenge_Protocol.md]: Oops. Shows I've had this wrong the whole time, haha.

**Comment by mcy** [comment]
Apparently I lied and I didn't have changes to push. Should all be addressed now, though!

**Review by chweimer — APPROVED** [review]

**Pull request merged**
